src/components/modal/header.js

```diff
@@ -15,9 +15,11 @@ const Header = forwardRef(({ classes, children, close, ...rest }, ref) => {
   return (
     <div ref={ref} className={`header-container ${modalClasses.header}`} {...rest}>
       <Dialog.Title as='span' className={modalClasses.title}>{children}</Dialog.Title>
-      <button className={modalClasses.close} onClick={close}>
-        <Close size='lg'/>
-      </button>
+      {!hideClose && 
```

Comment:
We can prob check for the presence of `close` supplied to the header or not, and toggle the close icon and its closing action based on that, that way we can reduce adding another prop `hideClose` just for that toggle

Reply:
will need a state which may cause doble re-render

Reply:
there's the `open` prop that specifies the open/closed state of the modal right?

Reply:
oh interesting 🤔, thought would be able to toggle open/close externally (e.g. set the `open` prop to false in the onClick handler of the "cancel" button)

Reply:
yes you can toggle it through cancel button if you take a look of the example in storybook the user controls it. the open prop. The modal component itself doesn't have a state management. The user is who controls it. This makes removing the X icon by the existence of closeModal prop not possible because closeModal must be passed in order to have the option to close the modal.

Reply:
Modal component itself pass down the closeModal prop to its Header automatically. What we can do here is instead of passing it down automatically when Modal.Header is used, the user has to pass it down by themselves in this way we can achieve what you suggested